BuildingExternalSources.md

```diff
@@ -0,0 +1,88 @@
+# Building sources from external directories
+
+If `X-HEEP` is vendorized into your project, you can still have software source files in any directory of your convenience and build them using `X-HEEP`'s compilation flow. 
```

Comment:
should we add some info regarding how the people can vendorize x-heep?

Reply:
I think its a good idea, but maybe not for this PR. 
The other day we had a long conversation with @benoitdenkinger about the "happiest" flow to follow during a (re)vendorization and it seems its worth giving it some thought.

Reply:
@benoitdenkinger @JuanSapriza can we speak about this? :)